services/workers/src/test/java/org/sagebionetworks/migration/worker/MigrationWorkerTest.java

```diff
@@ -2,105 +2,138 @@
+import org.sagebionetworks.repo.model.migration.RowMetadata;
+import org.sagebionetworks.repo.model.migration.RowMetadataResult;
 import org.sagebionetworks.schema.adapter.JSONObjectAdapter;
 import org.sagebionetworks.schema.adapter.JSONObjectAdapterException;
 import org.springframework.test.util.ReflectionTestUtils;
 
-import com.amazonaws.services.sqs.model.Message;
-
 public class MigrationWorkerTest {
 	
 	@Mock
 	private AsynchJobStatusManager mockAsynchJobStatusManager;
 	@Mock
 	private UserManager mockUserManager;
 	@Mock
-	private AsyncMigrationRequestProcessor mockRequestProcessor;
+	private MigrationManager mockMigrationManager;
 	@Mock
-	ProgressCallback<Void> mockProgressCallback;
+	private ProgressCallback mockCallback;
 	
+	ExecutorService migrationExecutorService;
 	MigrationWorker migrationWorker;
+	UserInfo user;
 	
 
 	@Before
 	public void setUp() throws Exception {
 		MockitoAnnotations.initMocks(this);
+		migrationExecutorService = Executors.newFixedThreadPool(2);
 		migrationWorker = new MigrationWorker();
 		ReflectionTestUtils.setField(migrationWorker, "asynchJobStatusManager", mockAsynchJobStatusManager);
 		ReflectionTestUtils.setField(migrationWorker, "userManager", mockUserManager);
-		ReflectionTestUtils.setField(migrationWorker, "requestProcessor", mockRequestProcessor);
-		
+		ReflectionTestUtils.setField(migrationWorker, "migrationExecutorService", migrationExecutorService);
+		ReflectionTestUtils.setField(migrationWorker, "migrationManager", mockMigrationManager);
+		user = new UserInfo(true);
 	}
 
 	@After
 	public void tearDown() throws Exception {
 	}
-
+	
 	@Test
-	public void testAsyncMigrationTypeCountRequest() throws Throwable {
-		String jobId = "1";
-		Message msg = new Message();
-		msg.setBody(jobId);
-		
-		AsynchronousJobStatus expectedJobStatus = new AsynchronousJobStatus();
-		expectedJobStatus.setJobId(jobId);
-		expectedJobStatus.setStartedByUserId(100L);
-		AsyncMigrationTypeCountRequest expectedReq = new AsyncMigrationTypeCountRequest();
-		expectedJobStatus.setRequestBody(expectedReq);
-		when(mockAsynchJobStatusManager.lookupJobStatus(anyString())).thenReturn(expectedJobStatus);
+	public void testProcessAsyncMigrationTypeCountRequest() throws Throwable {
+		AsyncMigrationTypeCountRequest mReq = new AsyncMigrationTypeCountRequest();
+		mReq.setType(MigrationType.ACCESS_APPROVAL.name());
 		
-		UserInfo expectedUserInfo = new UserInfo(true);
-		expectedUserInfo.setId(100L);
-		when(mockUserManager.getUserInfo(eq(100L))).thenReturn(expectedUserInfo);
+		migrationWorker.processAsyncMigrationRequest(mockCallback, user, mReq, "JOBID");
 		
-		migrationWorker.run(mockProgressCallback, msg);
+		verify(mockMigrationManager).processAsyncMigrationTypeCountRequest(user, mReq);
+		verify(mockAsynchJobStatusManager).setComplete(eq("JOBID"), any(AsynchronousResponseBody.class));
+	}
+	
+	@Test(expected=Exception.class)
+	public void testProcessAsyncMigrationTypeCountRequestFailed() throws Throwable {
```

Comment:
Since this applies to all request types, rename to "testProcessAsyncMigrationRequestFailed"?